test/beginend-dired-test.el

```diff
@@ -33,80 +33,87 @@
+
+(defun beginend-dired-test-begin (filename)
+  "Move the point to the first line of file list.
+Then, if FILENAME is non-nil, checks that point is at the beginning of
+FILENAME.  If FILENAME is nil, checks that point is at `point-max'."
+  (beginend-dired-test 'beginend-dired-mode-goto-beginning filename))
+
+(defun beginend-dired-test-end (filename)
+  "Move the point to the last line of file list.
+Then, if FILENAME is non-nil, checks that point is at the beginning of
+FILENAME.  If FILENAME is nil, checks that point is at `point-max'."
+  (beginend-dired-test 'beginend-dired-mode-goto-end filename))
+
+(defmacro with-beginend-dired-test-buffer (entries &rest body)
+  "Create a temporary directory with ENTRIES and execute BODY.
+
+ENTRIES is the file list.  The order is important.  ENTRIES are listed in that
+order.
+
+Execute the forms in BODY with the dired buffer of the directory."
+  (declare (indent 1))
+  `(with-current-buffer (beginend-create-dired-buffer ',entries)
+     ,@body))
 
 (describe "beginend in a dired buffer"
-  (describe "without dired-hide-details-mode"
-    (describe "without dired-omit-mode"
-      (it "offers a command to go to first file"
-        (beginend-dired-test 'begin-p))
-
-      (it "offers a command to go to last file"
-        (beginend-dired-test)))
-
-    (describe "with dired-omit-mode"
-      (it "offers a command to go to first file"
-        (beginend-dired-test 'begin-p 'omit))
-
-      (it "offers a command to go to last file with dired-omit-mode"
-        (beginend-dired-test 'omit))))
-
-  (describe "with-dired-hide-details-mode (also hides information lines)"
-    (describe "without dired-omit-mode"
-      (it "offers a command to go to first file"
-        (beginend-dired-test 'begin-p 'hide-details))
-
-      (it "offers a command to go to last file with dired-omit-mode"
-        (beginend-dired-test 'hide-details)))
-
-    (describe "with dired-omit-mode"
-      (it "offers a command to go to first file"
-        (beginend-dired-test 'begin-p 'omit 'hide-details))
-
-      (it "offers a command to go to last file with dired-omit-mode"
-        (beginend-dired-test 'omit 'hide-details))))
-
-  (describe "with-dired-hide-details-mode (does not hide information lines)"
-    (describe "without dired-omit-mode"
-      (it "offers a command to go to first file"
-        (beginend-dired-test 'begin-p 'hide-details 'keep-info-lines))
-
-      (it "offers a command to go to last file with dired-omit-mode"
-        (beginend-dired-test 'hide-details 'keep-info-lines)))
-
-    (describe "with dired-omit-mode"
-      (it "offers a command to go to first file"
-        (beginend-dired-test 'begin-p 'omit 'hide-details 'keep-info-lines))
-
-      (it "offers a command to go to last file with dired-omit-mode"
-        (beginend-dired-test 'omit 'hide-details 'keep-info-lines)))))
+  (it "ignores . and .. at the beginning"
+    (with-beginend-dired-test-buffer ("." ".." "dir1" "dir2")
```

Comment:
I think that

```emacs
(with-beginend-dired-test-buffer '("." ".." "dir1" "dir2")
  ...)
```

(with the quote) is easier to understand because it becomes clear that the parameter is not evaluated. Moreover, if the macro does not automatically quote its argument, users of the macro can build the macro parameter as a result of a function call or something else:

```emacs
(with-beginend-dired-test-buffer (a-function-that-builds-a-file-list)
  ...)
```